src/lib/validate-input.js

```diff
@@ -444,3 +444,17 @@ export function validateGatherParams(input, indices, {axis = 0} = {}) {
     }
   }
 }
+
+export function validateTriangularParams(input, {diagonal = 0} = {}) {
+  const inputRank = input.rank;
+  if (inputRank !== 2) {
+    throw new Error('The input should be a 2-D tensor.');
+  }
+  const [i, j] = input.shape;
+  if (i !== j) {
+    throw new Error('The input should be a 2-D tensor of [N, N] shape.');
+  }
+  if (diagonal >= i || diagonal <= -i) {
```

Comment:
Spec doesn't give the restrict for this **diagonal** . Please correct me, if I understand wrongly. Thanks.